### DISCUSSION TITLE
How to translate hardcoded component text

### DISCUSSION THREAD

**Discussion by leonardopugliese** [havit/Havit.Blazor#554] [Q&A] [answered]
Is it possible to customize the fields of the components in English?
Using for example I MessageBox Service MessageBox is there a way to have the localization of the text that is inserted in the buttons?
For example "Cancel" --> "Cancella", "Yes" --> "Si".
I've seen in code that many components use a resx to customize text. How to take care of localization and maybe share it with the community? Excuse me if I expressed myself in a confusing way

**Comment by hakenr** [comment, marked as answer]
Currently there are two ways to localization:
1. register your own `ILocalizedString<xy>` services to the dependency injection provider and provide the localizations you need,
2. create a pull-request to our library where you will add resources in your language
    * See the `Resources` folder here: https://github.com/havit/Havit.Blazor/tree/master/Havit.Blazor.Components.Web.Bootstrap/Resources
    * There are currently English (.resx), Czech (.cs.resx) and Chinese (.zh-CN.resx) resources.
    * Just create a copy of the English files, add extension referring to the language you are adding and translate the content, that's it.
    * There are only few strings to be localized, no major effort needed, your contribution will be more than welcomed!

**Reply by leonardopugliese** [reply]
I will be happy to give my contribution by adding the Italian language ;)
